pkg/nuctl/command/deploy.go

```diff
@@ -159,6 +161,18 @@ func newDeployCommandeer(rootCommandeer *RootCommandeer) *deployCommandeer {
 			commandeer.functionConfig.Meta.RemoveSkipDeployAnnotation()
 
 			commandeer.rootCommandeer.loggerInstance.DebugWith("Deploying function", "functionConfig", commandeer.functionConfig)
+			err = rootCommandeer.platform.EnrichFunctionConfig(&commandeer.functionConfig)
+			if err != nil {
+				return errors.Wrap(err, "Failed to enrich function config before deployment")
+			}
+
+			if _, ok := rootCommandeer.platform.(*kube.Platform); ok {
+				err = commandeer.overrideHTTPTrigger(&commandeer.functionConfig, v1.ServiceType(commandeer.overrideHTTPTriggerServiceType))
```

Comment:
```suggestion
				if err := commandeer.overrideHTTPTrigger(&commandeer.functionConfig, v1.ServiceType(commandeer.overrideHTTPTriggerServiceType)); err != nil {
```

Reply:
perhaps you can move this piece to
`func overrideFunctionConfig()`

and within you can have blocks that are platform kind oriented